engine/engine.go

```diff
@@ -112,9 +110,6 @@ func (e *EngineAPI) ForkchoiceUpdatedV3(
 		if err := e.builder.Rollback(fcs.HeadBlockHash, fcs.SafeBlockHash, fcs.FinalizedBlockHash); err != nil {
 			return nil, engine.GenericServerError.With(fmt.Errorf("rollback: %v", err))
 		}
-		if err := e.payloadStore.RollbackToHeight(headBlock.Header.Height); err != nil {
```

Comment:
Should we set the payload attrs to `nil` here?

Reply:
I don't think so.
 - `currentPayloadAttributes` ought to already be `nil`
 - the currently running function is carrying a `payloadAttributes` (`pa`) that will be used soon to populate it
 
The removed rollback call was presumably in place for circumstances where more involved payloadAttributes storage was being anticipated.